haystack/agents/base.py

```diff
@@ -131,9 +245,8 @@ def __init__(
         self,
         prompt_node: PromptNode,
         prompt_template: Union[str, PromptTemplate] = "zero-shot-react",
-        tools: Optional[List[Tool]] = None,
+        tools_manager: Optional[ToolsManager] = None,
```

Comment:
I wonder why do we expose the tool manager in the public API of the Agent? I would make the manager instance private to the Agent and only expose a method (or init  param, or both) to provide the tools:
```py
class Agent:
  def __init__(self,
    ..
    tools: Optional[List[Tool]]
    ..
  ):
    self._tool_manager = ToolManager(tools)     
     ...

  def add_tool(self, tool: Tool):
    self._tool_manager.add_tool(tool)
```

Reply:
Sure, will do, good idea @masci

Reply:
Or I remembered why - the reason is to allow users to specify the custom tool pattern on the level of ToolsManager. This pattern is not always the same.

Reply:
Ok, I guess we can add another parameter for the pattern of the tool to Agent init. LMK if you know some other better approach for the best trade-off we can get

Reply:
> Oh I remembered why - the reason is to allow users to specify the custom tool pattern on the level of ToolsManager. This pattern is not always the same.

Ok the old version was more future proof then. We need another param to the constructor anyways, it might as well be the ToolsManager itself.